### ISSUE TITLE
Protecting Raw routes.

### ISSUE THREAD

**Issue by nd2s** [plow-technologies/servant-auth#22]
I have a protected api type that defines a `Raw` route for websocket handling, like so:

```
type PrivApi =
         "foo" :> ReqBody '[JSON] Bar :> PostCreated '[JSON] Bar
    :<|> "foo" :> "stream" :> Raw

type Api =
         "login" :> ReqBody '[JSON] Creds :> Post '[JSON] Token
    :<|> Auth '[JWT] Session :> PrivApi

mkPrivApi :: AuthResult Session -> Server PrivApi
mkPrivApi (Authenticated s) = ...
mkPrivApi _                          = throwAll err401
```

The problem here is that `throwAll` does not work in this case:
```
    • Couldn't match type ‘GHC.IO.Exception.IOException’
                     with ‘ServantErr’
        arising from a functional dependency between:
          constraint ‘MonadError ServantErr IO’
            arising from a use of ‘throwAll’
          instance ‘MonadError GHC.IO.Exception.IOException IO’
            at <no location info>
    • In the expression: throwAll err401
      In an equation for ‘mkPrivApi’: mkPrivApi _ = throwAll err401
```

I tried to circumvent this by pulling the `Raw` route out of my `PrivApi` type (so that I can use `throwAll`) and putting it as an individually protected route in `Api`. But this lead to following error:
```
    • No instance for (HasServer
                         (Servant.Auth.Server.Internal.AddSetCookie.AddSetCookieApi Raw)
                         '[JWTSettings, CookieSettings])
        arising from a use of ‘serveWithContext’
```

Is there any way I can protect `Raw` routes like these?

**Comment by jkarni** [comment]
There are two things going on here - one is that the convenience function `throwAll` has to be generalized to cover the `Raw` case, and the other that `Raw` endpoints can't yet directly be protected.

The former is, I think, easy to fix, though not much point to doing so before fixing the latter. For the latter, I *think* it should be possible to make things work by having an `AddSetCookie` instance for `Application`, but I'll have to take a deeper look.

**Comment by jkarni** [comment]
@nd2s can you see if #24 fixes the second problem?

**Comment by nd2s** [comment]
I'm getting a new error now. It might have to do with the fact that `servant-auth-server` is now depending on a newer `servant-server` version:

```
    • No instance for (ToHttpApiData SetCookie)
        arising from a use of ‘serveWithContext’
    • In the first argument of ‘(.)’, namely
        ‘serveWithContext (Proxy :: Proxy Api) ctx’
```

It seems I'm getting this when using the `Auth` combinator (no `Raw` involved).

Unfortunately these error messages are not very helpful. I have no idea what `SetCookie` is.

**Comment by jkarni** [comment]
@nd2s sorry, that was a bug with 0.2.2.0. 0.2.3.0 should fix it

**Comment by jkarni** [comment]
Fixed in 0.2.4.0

**Comment by nd2s** [comment]
Seems to work nicely - many thanks!
